test/integration/point_cloud.cc

```diff
@@ -0,0 +1,204 @@
+#include <gz/msgs/world_stats.pb.h>
+
+#include <gz/common/Console.hh>
+#include <gz/common/Filesystem.hh>
+#include <gz/msgs/PointCloudPackedUtils.hh>
+#include <gz/msgs/Utility.hh>
+#include <gz/rendering/RenderEngine.hh>
+#include <gz/rendering/RenderingIface.hh>
+#include <gz/rendering/Scene.hh>
+#include <gz/transport/Node.hh>
+#include <gz/utils/ExtraTestMacros.hh>
+
+#include "test_config.hh"  // NOLINT(build/include)
+#include "gz/gui/Application.hh"
+#include "gz/gui/GuiEvents.hh"
+#include "gz/gui/MainWindow.hh"
+#include "gz/gui/Plugin.hh"
+
+int g_argc = 1;
+char* g_argv[] =
+{
+  reinterpret_cast<char*>(const_cast<char*>("./PointCloud_TEST")),
+};
+
+using namespace std::chrono_literals;
+
+using namespace gz;
+using namespace gui;
+
+class PointCloudTestFixture : public ::testing::Test
+{
+  public:
+    gz::transport::Node node;
+    rendering::ScenePtr scene;
+    gz::transport::Node::Publisher pointcloudPub;
+    gz::transport::Node::Publisher flatPub;
+    gz::msgs::PointCloudPacked pcMsg;
+    gz::msgs::Float_V flatMsg;
+    std::atomic<bool> receivedMsg;
+
+    PointCloudTestFixture()
+    {
+      // Periodic world statistics
+      this->pointcloudPub = this->node.Advertise<gz::msgs::PointCloudPacked>(
+        "/point_cloud");
+      this->flatPub = this->node.Advertise<gz::msgs::Float_V>("/flat");
+
+      this->InitMockData();
+
+      this->node.Advertise("/marker",
+        &PointCloudTestFixture::OnMarkerMsg, this);
+      this->receivedMsg = false;
+    }
+
+    public: void InitMockData()
+    {
+      gz::msgs::InitPointCloudPacked(pcMsg, "some_frame", true,
+          {{"xyz", gz::msgs::PointCloudPacked::Field::FLOAT32}});
+
+      int numberOfPoints{1000};
+      unsigned int dataSize{numberOfPoints * pcMsg.point_step()};
+      pcMsg.mutable_data()->resize(dataSize);
+      pcMsg.set_height(1);
+      pcMsg.set_width(1000);
+
+      // Populate messages
+      gz::msgs::PointCloudPackedIterator<float> xIter(pcMsg, "x");
+      gz::msgs::PointCloudPackedIterator<float> yIter(pcMsg, "y");
+      gz::msgs::PointCloudPackedIterator<float> zIter(pcMsg, "z");
+
+      for (float x = 0.0, y = 0.0, z = 0.0;
+          xIter != xIter.End();
+          ++xIter, ++yIter, ++zIter)
+      {
+        *xIter = x;
+        *yIter = y;
+        *zIter = z;
+        flatMsg.add_data(x);
+
+        x += 1.0;
+        if (x > 9)
+        {
+          x = 0.0;
+          y += 1.0;
+        }
+        if (y > 9)
+        {
+          y = 0.0;
+          z += 1.0;
+        }
+      }
+    }
+
+    public: void Publish()
+    {
+      this->pointcloudPub.Publish(pcMsg);
+      this->flatPub.Publish(flatMsg);
+    }
+
+    /// \brief Callback that receives marker messages.
+    /// \param[in] _req The marker message.
+    public: void OnMarkerMsg(const gz::msgs::Marker &_req)
+    {
+      if (_req.action() == gz::msgs::Marker::DELETE_ALL)
+      {
+        // TODO(arjo129): Don't clear all markers only one marker.
```

Comment:
Is this a feature `TODO` (ie. in the core code, not in this test)? If so, can you create an issue for better visibility?

Reply:
Ah this comment doesn't belong here.